discovery/discovery.go

```diff
@@ -148,12 +148,12 @@ func doDiscover(app App, pre string, insecure bool) (*Endpoints, error) {
 // DiscoverEndpoints will make HTTPS requests to find the ac-discovery meta
```

Comment:
@jonboulle Thinking more about it, I see some problems doing an unique DiscoverEndpoints:

*) The spec says that first simple discovery should be tried, then, if nothing can be downloaded, meta discovery should be tried.
For the same reasons explained in #114, if a simple discovery template is generated but there's a connection error (not 404 or other http errors), I think that meta discovery shouldn't be tried and the fetching process should stop here with an error.
So there's the need for the functions using discovery (like in rkt/fetch.go) to first get just the simple discovery urls, try to download them, and, based on the received error, try meta discovery or fail.

*) If an unique DiscoverEndpoints function is used it should combine simplediscovery url with metadiscovery url (putting simple ones first in the arrays). But if meta discovery fails for reasons explained in #114 it should return an error and so simple discovery won't be done (going against the spec).